src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -16,23 +16,31 @@
 
 package com.microsoft.hyperspace.index
 
-import org.apache.spark.sql.SparkSession
-import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import org.apache.spark.sql.types.{DataType, StructType}
 
+import com.microsoft.hyperspace.HyperspaceException
 import com.microsoft.hyperspace.actions.Constants
-import com.microsoft.hyperspace.index.serde.LogicalPlanSerDeUtils
+import com.microsoft.hyperspace.index.Content.Directory
 
 // IndexLogEntry-specific fingerprint to be temporarily used where fingerprint is not defined.
 case class NoOpFingerprint() {
   val kind: String = "NoOp"
   val properties: Map[String, String] = Map()
 }
 
+case class RelationMetadataEntry(
+    rootPaths: Seq[String],
+    files: Seq[String],
+    dataSchemaJson: String,
+    fileFormat: String)
```

Comment:
The source info from which index is created such as `dataSchemaJson`, `fileFormat`, etc. should be stored somewhere under `Source`. `Content` is a generic representation of directories (used by both `IndexLogEntry` and `Hdfs`, for example).

Reply:
Then how about "source" - `locations` - "content" - ...

other options for `locations`: baseRelation, baseDirectories, ..?